src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/aarch64/AArch64VaList.java

```diff
@@ -314,8 +309,8 @@ private Object read(Class<?> carrier, MemoryLayout layout, SharedUtils.Allocator
                         gpRegsArea.asSlice(currentGPOffset()));
                     consumeGPSlots(1);
 
-                    try (MemorySegment slice = handoffIfNeeded(ptr
-                            .asSegmentRestricted(layout.byteSize()), segment.ownerThread())) {
+                    try (ResourceScope scope = ResourceScope.ofConfined()) {
```

Comment:
not sure I get the suggestion, but when looking at this my feeling was that this method was just copying some data off to some other segment which is then allocated using the provided allocator. I believe what is missing here is a check that the guy calling read() can in fact do that (e.g. if the valist is confined, is the thread the same?) - but I'm not sure that, other than the check, we need something else?

Reply:
Yeah, AFAIK the only thing we need to check here is that the caller can actually read the segment. In fact, I don't think we need to create a scope here at all, since we don't return the created scope, we can just use the global scope I think.

Reply:
I think the right thing to do here is to use the VaList scope, not the global scope. That way, when reading from the va list segment we will be protected from read after close issues (since the slice we created would be backed by the same scope as the va list).